src/ape/managers/query.py

```diff
@@ -38,7 +38,14 @@ def perform_query(self, query: QueryType) -> pd.DataFrame:
 
     @perform_query.register
     def perform_block_query(self, query: BlockQuery) -> pd.DataFrame:
-        blocks_iter = self.chain_manager.blocks.range(query.start_block, query.stop_block)
+        if not self.network_manager.active_provider:
```

Comment:
It is not necessary to check if `network_manager.active_provider is not None`... Just use `self.provider` which automatically does this.

Reply:
Ah yes, please do this